src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2146,31 +2148,30 @@ std::error_code Session::receive_query_error_message(int error_code, std::string
 
 // The caller (Connection) must discard the session if the session has become
 // deactivated upon return.
-std::error_code Session::receive_error_message(int error_code, const ProtocolErrorInfo& info)
+std::error_code Session::receive_error_message(int error_code_int, const ProtocolErrorInfo& info)
 {
     logger.info("Received: ERROR \"%1\" (error_code=%2, try_again=%3, recovery_disabled=%4)", info.message,
-                error_code, info.try_again, info.client_reset_recovery_is_disabled); // Throws
+                error_code_int, info.try_again, info.client_reset_recovery_is_disabled); // Throws
 
     bool legal_at_this_time = (m_bind_message_sent && !m_error_message_received && !m_unbound_message_received);
     if (REALM_UNLIKELY(!legal_at_this_time)) {
         logger.error("Illegal message at this time");
         return ClientError::bad_message_order;
     }
 
-    bool known_error_code = bool(get_protocol_error_message(error_code));
+    bool known_error_code = bool(get_protocol_error_message(error_code_int));
     if (REALM_UNLIKELY(!known_error_code)) {
         logger.error("Unknown error code"); // Throws
         return ClientError::bad_error_code;
     }
-    ProtocolError error_code_2 = ProtocolError(error_code);
-    if (REALM_UNLIKELY(!is_session_level_error(error_code_2))) {
+    ProtocolError error_code = ProtocolError(error_code_int);
```

Comment:
I really don't like when we have variables that end in `_2`. This variable renaming is just to try to make the variable names match their purpose.